### PR TITLE
ROS-399 Port Ob.cc Primitives

### PR DESCRIPTION
**Pull request by notbdu**
Ports over the `object->string` primitive from the rosette vm. There is still quite a bit of work left to do but opening up PR for comments. 

Listing some TODOs:
- [x] Add test cases for `object->string`
- [x] Implement `object->symbol`
- [x] Add test cases for `object->symbol`

Unrelated questions:
- Need some help understanding how the `TypeMismatchMacro` works. I've tried looking up how `whitebox.Context`'s `macroApplication.filter` works (It's used here:https://github.com/rchain/rchain/pull/68/files#diff-1188bbd754795ad1979dd4dadf615c60R45) but I can't find anything except for a comment in the scala reflection docs [here](http://www.scala-lang.org/api/2.12.x/scala-reflect/scala/reflect/macros/blackbox/Context.html).

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/Ob.scala [hunk 1]
```diff
@@ -129,6 +129,9 @@ trait Ob extends Base with Cloneable {
 
   override def clone(): AnyRef = super.clone()
 
+  // Sub classes should implement this method
```
Comment (DoSOfRedRiver) [inline roscala/src/main/scala/coop/rchain/rosette/Ob.scala]: Does this change make sense?

File: roscala/src/main/scala/coop/rchain/rosette/prim/ob.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      Right(
+        RblString(
+          elem.foldLeft(init)((acc: String, el: Ob) => (acc ++ el.toString)))
+      )
+    }
+  }
+
+  /**
+    * Define the object->symbol primitive.
```
Comment (DoSOfRedRiver) [inline roscala/src/main/scala/coop/rchain/rosette/prim/ob.scala]: I think we could delete this commentary if you do not want to provide `object->symbol` primitive in that PR.

Comment (tymm) [reply roscala/src/main/scala/coop/rchain/rosette/prim/ob.scala]: Just implementing the primitive is enough. No need to implement a symbol table here.

**Comment by notbdu** [comment]
@tymm thanks for the feedback. Just pushed the changes you requested and the `objectToSymbol` primitive.

Reposting a question here for you:
- Need some help understanding how the `TypeMismatchMacro` works. I've tried looking up how `whitebox.Context`'s `macroApplication.filter` works (It's used here:https://github.com/rchain/rchain/pull/68/files#diff-1188bbd754795ad1979dd4dadf615c60R45) but I can't find anything except for a comment in the scala reflection docs [here](http://www.scala-lang.org/api/2.12.x/scala-reflect/scala/reflect/macros/blackbox/Context.html).

**Pull request merged**